DH-16

Dyrk84 opened the pull request:
championAccuracy() and monsterAccuracy() reworked in Fight class.
Implemented methods in champion and monster sides: critical miss, critical miss events, deadly hit, critical hit.
Reworked damage() methods in class of the characters.
Fight class methods reordered in running time.

Dyrk84 committed:
DH-16
championAccuracy() and monsterAccuracy() reworked in Fight class.
Implemented methods in champion and monster sides: critical miss, critical miss events, deadly hit, critical hit.
Reworked damage() methods in class of the characters.
Fight class methods reordered in running time.

Dyrk84 committed:
DH-16
Teszts reworking

Dyrk84 committed:
DH-16
Tests reworked.

Dyrk84 committed:
DH-16
main() reworked.
miss hit now activating, if finalAccuracy <= defense.

gergelyszanto commented on src/main/java/hu/dungeonhunter/characters/monsters/GoblinKing.java:
```diff
@@ -65,8 +72,9 @@ public boolean isDefeat() {
     @Override
-    public int getMonsterDamage() {
-        return Dice.rollDice(maxDamage, numOfDices);
+    public int Damage() {
```
Method names (with the exception of constructors) always start with lowercase.
This should be called: setDamage

gergely-szanto-epam committed:
DH-16 Refactor fight

Dyrk84 committed:
Merge branch 'master' of https://github.com/Dyrk84/dungeonhunter into DH-16

gergely-szanto-epam committed:
DH-16 fix stuffs

gergely-szanto-epam committed:
DH-16 fix stuffs

gergely-szanto-epam committed:
DH-16 add magyarázat :D

gergely-szanto-epam committed:
DH-16 improve initiation

Dyrk84 committed:
Merge remote-tracking branch 'origin/DH-16' into DH-16

gergelyszanto committed:
DH-16 fix stuffs

Dyrk84 committed:
Merge branch 'DH-16' of https://github.com/Dyrk84/dungeonhunter into DH-16

Dyrk84 committed:
DH-16
Some text for learning.
Better texts in good time in fight.
Fixed problems, if finalAccuracy is same.
Fixed problems on criticalMiss and dealDeadlyHit, that activated with rolledAccuracy, not with accuracyRoll. The rolledAccuracy renamed to finalAccuracy.
textOfWin reworked.
Fixed problem, the monster spawned in fight's turn middle. The monsters now incoming in nextTurn().

Dyrk84 committed:
DH-16
It is no longer possible to instantiate the hero as a character.
CharacterFactory is now MonsterFactory.
I started repairing the tests.
Champion isDefeated() repaired.

Dyrk84 committed:
DH-16
Some new tests repaired
Fight dealCriticalHit() error has been fixed

Dyrk84 committed:
DH-16
Some new tests repaired
monsterDefeated() if GoblinKing defeated, monsterCounter don't decreased.

Dyrk84 committed:
DH-16
Textseparator fixed
RuntimeException Test fixed

gergelyszanto commented on src/test/java/hu/dungeonhunter/TestFight.java:
```diff
@@ -169,397 +179,302 @@ public void monsterCallerGoblinTest() {
+        fight.attack(monster, 1);
 
-        fight.battle();
+        softly.assertThat(fight.getRandomCriticalMissEvent()).isLessThan(5);
```
Don't use soft assertion if there is only a single assertion in the test.

Dyrk84 replied:
fixed

gergelyszanto commented on src/test/java/hu/dungeonhunter/TestFight.java:
```diff
@@ -169,397 +179,302 @@ public void monsterCallerGoblinTest() {
-        softly.assertThat(fight.getMonsterCounter()).isEqualTo(startingMonsterCounter);
-        softly.assertThat(fight.getChampion().getHealingPotionCounter()).isEqualTo(startingHealingPotionCounter);
-        softly.assertThat(fight.nextTurn()).isTrue();
+        softly.assertThat(fight.getRandomCriticalMissEvent()).isLessThan(5);
         softly.assertAll();
```
Same here, no need for softly asserting.

Dyrk84 replied:
Thank you, fixed!

gergelyszanto commented on src/test/java/hu/dungeonhunter/TestFight.java:
```diff
@@ -169,397 +179,302 @@ public void monsterCallerGoblinTest() {
+        monster.setHp(1);
         fight.setMonster(monster);
+        fight.setRandomCriticalMissEvent(1);
+        int[] monsterHpAfterDamage = {fight.getMonster().getHp() - fight.getMonster().getMaxDamage(),
+                fight.getMonster().getHp() - fight.getMonster().getNumOfDices()};
```
I believe this is incorrect, as you don't count in the number of dices to the may damage.
```suggestion
        int[] monsterHpAfterDamage = {fight.getMonster().getHp() - fight.getMonster().getMaxDamage() * fight.getMonster().getNumOfDices(),
                fight.getMonster().getHp() - fight.getMonster().getNumOfDices()};
```

Dyrk84 replied:
Its true! I fixed this fail for all these types of  tests.

Dyrk84 committed:
DH-16
some tests fixed
